server/src/api/invitations/invitations.controller.ts

```diff
@@ -34,15 +44,28 @@ export const postInvitation: RouteHandlerFnc = async (ctx) => {
 
   if (existingRole) throw new Error('Invitee is already a team member');
 
-  const created = await Invitation.query()
-    .insertAndFetch({
-      id: uuid.v4(),
-      roadmapId,
-      type,
-      email,
-    })
-    .onConflict(['roadmapId', 'email'])
-    .merge();
+  const created = await Invitation.transaction(async (trx) => {
+    const invitation = await Invitation.query(trx)
+      .insertAndFetch({
+        id: uuid.v4(),
+        roadmapId,
+        type,
+        email,
+      })
+      .onConflict(['roadmapId', 'email'])
+      .merge();
+
+    if (representativeFor?.length > 0) {
+      const customers = await Customer.query(trx)
```

Comment:
> Also, it would be good to check that we found all the requested customers.

Should we return status 400 if that's not the case?

Reply:
400 would indicate bad request, but the request would be ok (correct fields etc.) in this case.
Maybe 404, since some of the requested resources could not be found.

Reply:
Yeah 404 seems to be more fitting.